src/FrameMapper.cpp

```diff
@@ -785,10 +785,23 @@ void FrameMapper::ResampleMappedAudio(std::shared_ptr<Frame> frame, int64_t orig
 	// Create a new array (to hold all S16 audio samples for the current queued frames)
  	int16_t* frame_samples = (int16_t*) av_malloc(sizeof(int16_t)*total_frame_samples);
 
-	// Translate audio sample values back to 16 bit integers
-	for (int s = 0; s < total_frame_samples; s++)
-		// Translate sample value and copy into buffer
-		frame_samples[s] = int(frame_samples_float[s] * (1 << 15));
+	// Translate audio sample values back to 16 bit integers with saturation
+	float valF;
+	int16_t conv;
+	const int16_t max16 = 32767;
+	const int16_t min16 = -32768;
+	for (int s = 0; s < total_frame_samples; s++) {
```

Comment:
Wait, there are **two** places where this exact same conversion is done!?

...Why the heck does `openshot::Frame::GetInterleavedSamples()` return a float array, then? Instead of at least having an overload that converts it first, and passes back an array in the format we actually need!? ARRRGH, I hate cut-and-paste code duplication.

Well, anyway, that part's not affected by these changes, so not really relevant here.